### PR TITLE
doc: emphasize that the finish should be called only when the output …

### PR DESCRIPTION
**Pull request by spacewander**
…is finished

### CODE REVIEW

**Comment by wu-sheng** [comment]
@moonming Please take a look. When this wouldn't be `eof`? We may need more checks if there is another case.

File: README.md [hunk 1]
```diff
@@ -48,7 +48,9 @@ http {
             proxy_pass http://127.0.0.1:8080/backend;
 
             body_filter_by_lua_block {
-                require("tracer"):finish()
+                if ngx.eof then
```
Comment (wu-sheng) [inline README.md]: @spacewander Could you apply this change to https://github.com/apache/skywalking-nginx-lua/blob/master/examples/nginx.conf#L78 too?

That file is for local closed-loop test, it makes the dev team doesn't require the SkyWalking backend.

Comment (moonming) [reply README.md]: @spacewander is `ngx.eof` the same as `ngx.arg[2]`?  Is it a hidden variable?

**Comment by moonming** [comment]
> @moonming Please take a look. When this wouldn't be `eof`? We may need more checks if there is another case.

the `eof` means the end of the response body data

File: README.md (hunk 1 above)
Comment (spacewander) [reply README.md]: @wu-sheng @moonming 
Sorry, it's my mistake. It should be `ngx.arg[2]` indeed.

Comment (spacewander) [reply README.md]: Fixed in the new commit.

Comment (wu-sheng) [reply README.md]: What is `ngx.arg[2]`? Could you explain a little more? If `the eof means the end of the response body data`, why isn't it suitable?

Comment (spacewander) [reply README.md]: @wu-sheng 
`ngx.eof` is a function which marks the end of the body, but not the mark.
In `body_filter_by_lua*` phase, the `ngx.arg[1]` returns the body data and `ngx.arg[2]` returns the mark itself (a metamethod is called under the hover).  
So in my previous commit, use `ngx.eof` (a function) is incorrect, `ngx.arg[2]`(the mark) should be used instead.

**Review by wu-sheng — APPROVED** [review]
LGTM. @moonming Agree?

**Comment by moonming** [comment]
yes, LGTM

**Review by moonming — APPROVED** [review]

**Pull request merged**